core/base/inc/TNamed.h

```diff
@@ -33,7 +33,7 @@ class TNamed : public TObject {
    TString   fTitle;           //object title
 
 public:
-   TNamed(): fName(), fTitle() { }
+   TNamed(): fName(), fTitle() { } // NOLINT: not allowed to use = default because of TObject::kIsOnHeap detection, see ROOT-10300
```

Comment:
```suggestion
   TNamed() {} // NOLINT: not allowed to use = default. See note on \ref TObject
```

Maybe drop the other initializers (they should be handled implicitly, I hope)?
Maybe reference the just newly added docs instead of something somewhere else?